python/ray/serve/tests/unit/test_schema.py

```diff
@@ -726,5 +728,52 @@ def f():
     assert deployment.ray_actor_options["num_cpus"] == 1
 
 
+def test_serve_instance_details_is_json_serializable():
+    """Test that ServeInstanceDetails is json serializable."""
+    serialized_policy_def = (
+        b"\x80\x05\x95L\x00\x00\x00\x00\x00\x00\x00\x8c\x1cray."
+        b"serve.autoscaling_policy\x94\x8c'replica_queue_length_"
+        b"autoscaling_policy\x94\x93\x94."
+    )
+    details = ServeInstanceDetails(
+        controller_info={"node_id": "fake_node_id"},
+        proxy_location="EveryNode",
+        proxies={"node1": {"status": "HEALTHY"}},
+        applications={
+            "app1": {
+                "name": "app1",
+                "route_prefix": "/app1",
+                "docs_path": "/docs/app1",
+                "status": "RUNNING",
+                "message": "fake_message",
+                "last_deployed_time_s": 123,
+                "deployments": {
+                    "deployment1": {
+                        "name": "deployment1",
+                        "status": "HEALTHY",
+                        "status_trigger": "AUTOSCALING",
+                        "message": "fake_message",
+                        "deployment_config": {
+                            "name": "deployment1",
+                            "autoscaling_config": {
+                                # Byte object will cause json serializable error
+                                "serialized_policy_def": serialized_policy_def
+                            },
+                        },
+                        "replicas": [],
+                    }
+                },
+            }
+        },
+    )._get_user_facing_json_serializable_dict(exclude_unset=True)
+    json.dumps(details)
+
+    # ensure internal field, serialized_policy_def, is not exposed
+    application = details["applications"]["app1"]
```

Comment:
nit: the test case name is checking json serialization, suggest to add expected result and do strict compare between json output and expected output, so that we can prevent other fields in the future.

Reply:
I feel the test should only check the behavior we expect, in this case the output should be json serializable and not containing internally field (serialized_policy_def). Checking the exact output would be too rigid and hard to maintain. If not a strong preference I would keep this as it :)